src/test/unit/model/indexing/rvalue_cl_test.cpp

```diff
@@ -0,0 +1,273 @@
+#ifdef STAN_OPENCL
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+#include <stan/model/indexing/rvalue.hpp>
+#include <stan/model/indexing/rvalue_cl.hpp>
+#include <stan/math.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/util.hpp>
+#include <test/unit/pretty_print_types.hpp>
+#include <tuple>
+
+using stan::model::rvalue;
+
+using stan::model::index_max;
+using stan::model::index_min;
+using stan::model::index_min_max;
+using stan::model::index_multi;
+using stan::model::index_omni;
+using stan::model::index_uni;
+
+template <typename T>
+T opencl_index(T i) {
+  return i;
+}
+stan::math::matrix_cl<int> opencl_index(const index_multi& i) {
+  return stan::math::to_matrix_cl(i.ns_);
+}
+
+template <typename T>
+auto from_matrix_cl_nonscalar(const T& a) {
+  return stan::math::from_matrix_cl(a);
+}
+auto from_matrix_cl_nonscalar(const stan::math::var a) { return a; }
+auto from_matrix_cl_nonscalar(const double a) { return a; }
+auto from_matrix_cl_nonscalar(const int a) { return a; }
+
+template <typename T>
+void set_adjoints(T& v) {
+  for (int i = 0; i < v.rows(); i++) {
+    for (int j = 0; j < v.cols(); j++) {
+      v(i, j).adj() += i + 3;
+    }
+  }
+}
```

Comment:
Can you make a `util` file to put these in with the things used in the assign test as well?

Reply:
I feel like we already have many util files scattered around. I always have to check what is in which one. So I don't think making another one would improve clarity.

Reply:
idt it's a big deal to have util files around, and we can just use another of the util files as well. It's mostly that the functions here and in the other file are very similar so I don't see a reason to have two sets of them.

Reply:
fine